Libraries/LibC/netdb.cpp

```diff
@@ -468,7 +454,7 @@ struct protoent* getprotoent()
     if (!fill_getproto_buffers(line, read))
         return nullptr;
 
-    __getproto_buffer.p_name = __getproto_name_buffer;
+    __getproto_buffer.p_name = const_cast<char*>(__getproto_name_buffer.characters());
```

Comment:
These const casts everywhere are somewhat unfortunate.

For this commit, adding them is fine: the API requires returning/offering a `char*`.

But maybe in another commit/PR, we could "lock down" the API (here and everywhere else) and mark the fields in the structs as `const char*`s? I hope this only produces a warning in ports that copy the point into a `char* foo;`, and not an error. And within Serenity, this would remove a lot of `const-cast`s.

Reply:
It is unfortunate, but that's Unix... The API isn't beautiful in other aspects either.

I... don't know, I think I'd rather `const_cast<>` it and keep the API as-is.

In C++, it's surely an error to implicitly cast a `const char` to `char*`. Do we have any C++ ports? But then again, they're ports, it's fine to patch them if that's required to make them build/work...